strawberryfields/parameters.py

```diff
@@ -19,335 +19,334 @@
   using :meth:`~ops.Operation.decompose`, and finally merging and commuting Commands inside
   the graph representing the quantum circuit.
   The circuit graph is built using the knowledge of which subsystems the Commands act and depend on.
 
-* Merging two :class:`.Gate` instances of the same subclass involves
+* Decompositions, merges and commutations often involve creation new Operations with algebraically
+  transformed parameters.
+  For example, merging two :class:`.Gate` instances of the same subclass involves
   adding their first parameters after equality-comparing the others. This is easily done if
   all the parameters have an immediate numerical value.
-  RegRefTransforms and TensorFlow objects are more complicated, but could in principle be handled.
-  For now, we simply don't merge Operations that depend on RegRefTransforms or TensorFlow objects.
+  Measured and free parameters are handled symbolically by Sympy.
 
 * The compiled Program is run by a :class:`.BaseEngine` instance, which calls the
-  :meth:`~ops.Operation.apply` method of each Operation in turn.
+  :meth:`apply` method of each Operation in turn.
 
-* :meth:`Operation.apply` evaluates the numeric value of any RegRefTransform-based Parameters
-  using :meth:`Parameter.evaluate` (other types of Parameters are simply passed through).
-  The parameter values and the subsystem indices are passed to :meth:`Operation._apply`.
-
-* :meth:`~ops.Operation._apply` "unwraps" the Parameter instances. There are three different cases:
-
-  1. We still need to do some arithmetic, unwrap after it is done using `p.x`.
-  2. No arithmetic required, use :func:`~parameters._unwrap`.
-  3. No parameters are used, do nothing.
-
-  Finally, :meth:`_apply` calls the appropriate backend API method using the unwrapped parameters.
+* :meth:`apply` then calls :meth:`_apply` which is unique for each Operation subclass.
+  It may perform additional numeric or symbolic transformations on the parameters, and
+  then finally evaluates the numeric value of the parameters using :func:`par_evaluate`.
+  The numeric parameter values are passed to the appropriate backend API method.
   It is up to the backend to either accept NumPy arrays and Tensorflow objects as parameters, or not.
 
+
 What we cannot do at the moment:
 
-* Use anything except integers and RegRefs (or Sequences thereof) as the subsystem parameter
+* Use anything except integers and RegRefs (or Sequences thereof) as the subsystem argument
   for the :meth:`~ops.Operation.__or__` method.
-  Technically we could allow any Parameters or valid Parameter initializers that evaluate into an integer.
-* Do arithmetic with RegRefTransforms.
+  Technically we could allow any parameters that evaluate into an integer.
 
-Parameter methods
+
+Functions
+---------
+
+.. currentmodule:: strawberryfields.parameters
+
+.. autosummary::
+   par_evaluate
+   par_is_symbolic
+   par_regref_deps
+   par_str
```

Comment:
Very minor comment, but do these need to all start with `par`? Being in the parameter module/namespace, it should be implicit

Reply:
:+1:

Reply:
I wanted to make the context explicit so it's easy to search the function names from the codebase (evaluate and str would otherwise show up a lot)